src/types/list.rs

```diff
@@ -750,6 +807,27 @@ impl<'py> Iterator for BoundListIterator<'py> {
             None
         })
     }
+
+    #[inline]
+    #[cfg(all(Py_GIL_DISABLED, feature = "nightly"))]
+    fn advance_by(&mut self, n: usize) -> Result<(), NonZero<usize>> {
+        self.with_critical_section(|index, length, list| {
+            let length = length.0.min(list.len());
+            let target_index = index.0 + n;
+            if index.0 + n < length {
+                let item = list.get_item(target_index);
```

Comment:
I don't think we need to actually do the `get_item` or check for an `Err` value, we only need to advance the iterator, which just requires the `index.0 + n < length` check.